stubs/common/Attribute.stub

```diff
@@ -27,7 +27,7 @@ class Attribute
      *
      * @template TMakeGet
      * @template TMakeSet
-     * @param  (callable(mixed, mixed): TMakeGet)|null  $get
+     * @param  (callable(mixed, array): TMakeGet)|null  $get
      * @param  (callable(TMakeSet, mixed=): mixed)|null  $set
```

Comment:
You should also adjust the other types (including the `get` and `set` methods below):

```suggestion
     * @param  (callable(TMakeSet, array<string, mixed>=): mixed)|null  $set
```

Reply:
This may just be due to a lack of knowledge on my part, but if you specify the array type in detail, problems may arise when you actually use it.
I don't know how to get around this easily.

```php
    /**
     * @return Attribute<Period, Period>
     */
    protected function publishPeriod(): Attribute
    {
        return Attribute::make(
            get: fn(mixed $value, array $attributes) => new Period(
                $attributes['publish_from'],
                $attributes['publish_to'],
            ),
            set: fn(Period $value) => [
                'publish_from' => $value->from,
                'publish_to' => $value->to,
            ],
        );
```
```
 ------ -------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Models/Recruitment.php                                                                                                                      
 ------ -------------------------------------------------------------------------------------------------------------------------------------------- 
  24     Parameter $get of static method Illuminate\Database\Eloquent\Casts\Attribute<mixed,mixed>::make() expects (callable(mixed, array<string,    
         mixed>=): App\ValueObjects\Period)|null, Closure(mixed, array): App\ValueObjects\Period given.                                              
         🪪  argument.type                                                                                                                           
         💡 Type #1 from the union: Parameter #2 $attributes of passed callable is required but the parameter of accepting callable is optional. It  
            might be called without it.                                                                                                              
  25     Parameter #1 $from of class App\ValueObjects\Period constructor expects string, mixed given.                                                
         🪪  argument.type                                                                                                                           
  26     Parameter #2 $to of class App\ValueObjects\Period constructor expects string, mixed given.                                                  
         🪪  argument.type                                                                                                                           
 ------ -------------------------------------------------------------------------------------------------------------------------------------------- 
````

Reply:
Ah, this is due to the `=` on the end of the array---this is to designate that the calling code might not pass it not not that the called code might not use it. You can just remove the `=` so that you get `array<string, mixed>`

Reply:
get: https://github.com/laravel/framework/blob/11.x/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php#L706-L708

set:
https://github.com/laravel/framework/blob/11.x/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php#L1854

Reply:
I still haven't figured out a way to avoid this.

I'm very sleepy, so I'll get back to it another day. (I have plans tomorrow, so it will be after the day after tomorrow).

```
 25     Parameter #1 $from of class App\ValueObjects\Period constructor expects string, mixed given.                                                
         🪪  argument.type                                                                                                                           
  26     Parameter #2 $to of class App\ValueObjects\Period constructor expects string, mixed given.                                                  
         🪪  argument.type  
```

Reply:
That's because PHPStan knows nothing about the array shape and contents. You can cast to a string for now


```php
            get: fn(mixed $value, array $attributes) => new Period(
                (string) $attributes['publish_from'],
                (string) $attributes['publish_to'],
            ),
```

Reply:
Yeah, I tried using Cast, but I still got an error for that.😭

```
 ------ ------------------------------ 
  Line   Models/Recruitment.php        
 ------ ------------------------------ 
  26     Cannot cast mixed to string.  
         🪪  cast.string               
  27     Cannot cast mixed to string.  
         🪪  cast.string               
 ------ ------------------------------ 
 ```

Reply:
I wish I could do 'as ~' like in TypeScript...

```
            get: fn(mixed $value, array $attributes) => new Period(
                $attributes['publish_from'] as string,
                $attributes['publish_to'] as string,
            ),
```

Reply:
that wouldn't change anything and ain't how casting works in php

Reply:
I can't think of a good way to do this, so I'm going to ignore it in my project.
```
                $attributes['publish_from'], // @phpstan-ignore-line
                $attributes['publish_to'], // @phpstan-ignore-line
```

Thank you for your cooperation.

Reply:
or just add a global `Cannot cast mixed` phpstan ignore